public/src/pages/views/UserView.tsx

```diff
@@ -75,22 +74,23 @@ export class UserView extends React.Component<IUserViewerProps, IUserViewerState
         return heading;
     }
 
-    private renderRow(user: IUserRelation): (string | JSX.Element)[] {
+    private renderRow(enr: Enrollment): (string | JSX.Element)[] {
         const selector: (string | JSX.Element)[] = [];
-        if (user.enrollment.getStatus() === Enrollment.UserStatus.TEACHER) {
+        const user = enr.getUser();
+        if (user && enr.getStatus() === Enrollment.UserStatus.TEACHER) {
```

Comment:
This code can be further simplified by checking for `!user` and returning an empty selector early (which should probably never happen, right??). After which you don't need to check for `user` again, and you can also avoid the `end.getUser()?` checks below. Like this:

```typescript
        if (!user) {
            return selector;
        }
        if (enr.getStatus() === Enrollment.UserStatus.TEACHER) {
            selector.push(
                <span className="text-muted">
                    <a href={this.gitLink(user.getLogin())} target="_blank">{user.getName()}</a>
                </span>);
        } else {
            selector.push(
                <a href={this.repoLink(user.getLogin())} target="_blank">{user.getName()}</a>);
        }
        selector.push(
            <a href={"mailto:" + user.getEmail()}>{user.getEmail()}</a>, user.getStudentid() ?? "",
        );
```

Reply:
Changed.